Test Module00a

pdiakumis opened the pull request:
Testing GATK-SV's Module00a subworkflow and individual tasks.

@illusional files are split into SVGs from `womtool graph`, individual workflow WDLs (mostly simplified/cleaned from the official repo), general input jsons (with docker/refdata files) and analysis-runner scripts (with sample inputs and links to general input jsons).

illusional commented on gatk-sv/test/Module00a/CollectCoverage/run.sh:
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+curl --location \
```
```suggestion
# Runs WDL workflow using analysis-runner
curl --location \
```

illusional commented on gatk-sv/test/Module00a/CollectCoverage/workflow.wdl:
```diff
@@ -0,0 +1,30 @@
+version 1.0
+
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/Structs.wdl"
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/CollectCoverage.wdl" as cov
```
Although this is convenient, linking to an external resource makes for a less confident approve for running in the analysis runner, because:
- we're relying on an external resource to run our workflows. 
- tags can change

Given it's the same repository, can't we merge changes from upstream?

Although I still don't feel great about approving the _soft_ link to an external resource, I'd budge for linking to the commit hash. (But I'd still prefer performing a merge from upstream)

Alternative:
```suggestion
# from gatk-sv/releases/v0.15-beta
import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/f64d5c8b808eed7ba470132d84a86229c4c2cfeb/wdl/Structs.wdl"
import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/f64d5c8b808eed7ba470132d84a86229c4c2cfeb/wdl/CollectCoverage.wdl" as cov
```

illusional commented on gatk-sv/test/Module00a/Delly/run.sh:
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+curl --location \
```
```suggestion
# Run delly workflow in cromwell through the analysis-runner
curl --location \
```

illusional commented on gatk-sv/test/Module00a/Manta/run.sh:
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+curl --location \
```
```suggestion
# run manta through cromwell in the analysis runner
curl --location \
```

illusional commented on gatk-sv/test/Module00a/Manta/workflow.wdl:
```diff
@@ -0,0 +1,40 @@
+version 1.0
+
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/Structs.wdl"
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/Manta.wdl" as manta
```
See above comment.

illusional commented on gatk-sv/test/Module00a/PESRCollection/run.sh:
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+curl --location \
```
```suggestion
# Run PERSCollection workflow in cromwell through the analysis runner
curl --location \
```

illusional commented on gatk-sv/test/Module00a/PESRCollection/workflow.wdl:
```diff
@@ -0,0 +1,35 @@
+version 1.0
+
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/Structs.wdl"
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/PESRCollection.wdl" as PESRC
```
See above comment.

```suggestion
# from gatk-sv/releases/v0.15-beta
import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/f64d5c8b808eed7ba470132d84a86229c4c2cfeb/wdl/Structs.wdl"
import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/f64d5c8b808eed7ba470132d84a86229c4c2cfeb/wdl/PESRCollection.wdl" as PESRC
```

illusional commented on gatk-sv/test/Module00a/Whamg/run.sh:
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+curl --location \
```
```suggestion
# run Whamg workflow in cromwell through the analysis runner
curl --location \
```

illusional commented on gatk-sv/test/Module00a/Whamg/workflow.wdl:
```diff
@@ -0,0 +1,83 @@
+version 1.0
+
+import "../../gatk-sv-git/wdl/Whamg.wdl" as wham
+import "../../gatk-sv-git/wdl/Structs.wdl"
+import "../../gatk-sv-git/wdl/CramToBam.wdl" as ctb
```
These three imports are linked to a git-ignored directory.

illusional commented on gatk-sv/test/Module00a/run.sh:
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+curl --location \
```
```suggestion
# run Module00a in cromwell through the analysis runner
curl --location \
```

illusional commented on gatk-sv/test/Module00a/workflow.wdl:
```diff
@@ -0,0 +1,167 @@
+version 1.0
+
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/CollectCoverage.wdl" as cov
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/CramToBam.wdl" as ctb
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/Delly.wdl" as delly
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/Manta.wdl" as manta
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/PESRCollection.wdl" as pesr
+import "https://raw.githubusercontent.com/broadinstitute/gatk-sv/v0.15-beta/wdl/Whamg.wdl" as wham
```
See above comment.

illusional reviewed (changes requested):
Thanks @pdiakumis! You're importing a specific release of the parent fork of gatk-sv, I've left a comment on the first usage discussing this, but it sounds like we should be merging from upstream.

pdiakumis commented:
Thanks for the review @illusional!
I added back the gatk-sv submodule and am now linking to it in the import statements. I've also added comments in the runner scripts. Let me know if that makes more sense.

illusional reviewed (approved):
Thanks for making those changes, looks great! I see it's `gatk_sv` is pinned to [`9a79c89`](https://github.com/populationgenomics/gatk-sv/tree/9a79c89f616580204aa5b4aed3b365b233e92a2d) which looks right.
